paddle/phi/core/flags.h

```diff
@@ -24,34 +24,141 @@
 
 #include "paddle/utils/variant.h"
 
+#if defined(_WIN32)
+#if defined(BUILD_PHI_SHARED)
+#define PHI_EXPORT_FLAG __declspec(dllexport)
+#define PHI_IMPORT_FLAG __declspec(dllimport)
+#else
+#define PHI_EXPORT_FLAG
+#define PHI_IMPORT_FLAG
+#endif
+#else
+#define PHI_EXPORT_FLAG
+#define PHI_IMPORT_FLAG
+#endif  // _WIN32
```

Comment:
Maybe the following change can make codes more clear?

```
#if defined(_WIN32) && defined(BUILD_PHI_SHARED)
#define PHI_EXPORT_FLAG __declspec(dllexport)
#define PHI_IMPORT_FLAG __declspec(dllimport)
#else
#define PHI_EXPORT_FLAG
#define PHI_IMPORT_FLAG
#endif
```

Reply:
done,3q